R/plots.R

```diff
@@ -274,17 +228,46 @@ use_regular_grid_plot <- function(x) {
 
 # ------------------------------------------------------------------------------
 
-plot_perf_vs_iter <- function(x, metric = NULL, width = NULL, eval_time = NULL) {
+process_autoplot_metrics <- function(x, metric, eval_time) {
+  x <- estimate_tune_results(x)
+  # This next line updates the .metric columns to be consistent with what the
```

Comment:
This took a few min to figure out so I documented here (should it by anywhere else)?

Reply:
Maybe move the documentation to to `paste_param_by()`? this way we don't have to copy it around if we need to use `paste_param_by()` somewhere else

Reply:
I think that this is a good idea and @simonpcouch could do that in a separate PR 😄

Reply:
#835